spimdisasm/mips/symbols/MipsSymbolBase.py

```diff
@@ -193,36 +193,32 @@ def shouldMigrate(self) -> bool:
         return False
 
 
-    def renameBasedOnType(self):
-        if not common.GlobalConfig.AUTOGENERATED_NAMES_BASED_ON_DATA_TYPE:
-            return
-
+    def autogenerateName(self, contextSym, inSymbolOffset):
         if not self.contextSym.isAutogenerated:
             return
 
-        if self.isFloat(0):
-            self.contextSym.name = f"FLT_{self.vram:08X}"
-        elif self.isDouble(0):
-            self.contextSym.name = f"DBL_{self.vram:08X}"
-        elif self.isString():
-            self.contextSym.name = f"STR_{self.vram:08X}"
-        elif self.isPascalString():
-            self.contextSym.name = f"PSTR_{self.vram:08X}"
-
+        if common.GlobalConfig.AUTOGENERATED_NAMES_BASED_ON_FILE_NAME and self.parent is not None:
+            offset = self.inFileOffset + inSymbolOffset
+            contextSym.name = f"{self.parent.name}{self.sectionType.toStr()}_{offset:06X}"
```

Comment:
I think it would be nice to still know this symbol is a data symbol
```suggestion
            contextSym.name = f"D_{self.parent.name}{self.sectionType.toStr()}_{offset:06X}"
```

Reply:
Also, would `self.sectionType.toStr()` return a name with a dot, right? Like `.data`. A symbol like that couldn't be used on C, which would make this feature useless for any decomp repo

Reply:
Yeah I thought this feature would be used by humans to compare assembly diffs, so I optimized for readability (something like `audio_data_data_000000` could be confusing). I'm not sure you'd want this when decomping the first version of a game anyway though, these names could change if your data splits are wrong but addresses are more stable